sos/plugins/networking.py

```diff
@@ -251,8 +227,15 @@ def setup(self):
             # Devices that exist in a namespace use less ethtool
             # parameters. Run this per namespace.
             for namespace in self.get_ip_netns(ip_netns_file):
-                for eth in self.get_netns_devs(namespace):
-                    ns_cmd_prefix = cmd_prefix + namespace + " "
+                ns_cmd_prefix = cmd_prefix + namespace + " "
+                netns_netdev_list = self.call_ext_prog(ns_cmd_prefix +
+                                                       "ls -1 /sys/class/net/")
```

Comment:
Is there a reason not to use `os.listdir()` here? It is typically cleaner than shelling out to `ls`...

Reply:
Ah, OK. The ns prefix. I wonder if there isn't a better way to do that, it's kinda fugly...

Reply:
Yeah, haven't found a direct pythonic way outside calling shell. I can add a comment if it clarifies the code..?